src/pcre2_jit_compile.c

```diff
@@ -13223,7 +13223,7 @@ common->compiler = compiler;
 
 /* Main pcre2_jit_exec entry. */
 SLJIT_ASSERT((private_data_size & (sizeof(sljit_sw) - 1)) == 0);
-sljit_emit_enter(compiler, 0, SLJIT_ARGS1(W, W), 5 | SLJIT_ENTER_FLOAT(SLJIT_NUMBER_OF_SCRATCH_FLOAT_REGISTERS), 5, private_data_size);
+sljit_emit_enter(compiler, 0, SLJIT_ARGS1(W, W), 5 | SLJIT_ENTER_VECTOR(SLJIT_NUMBER_OF_SCRATCH_VECTOR_REGISTERS), 5, private_data_size);
```

Comment:
is SLJIT SIMD really being used now, or this is an abstraction that leaked?

Reply:
The x86 uses the simd interface. Now the simd interface has correct vector register handling.

I did not have time to merge the others and simplify those platforms.